x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/create_rules_bulk_route.ts

```diff
@@ -90,6 +91,8 @@ export const createRulesBulkRoute = (router: IRouter) => {
             } = payloadRule;
             const ruleIdOrUuid = ruleId ?? uuid.v4();
             try {
+              validateLicenseForRuleType({ license: context.licensing.license, ruleType: type });
```

Comment:
Is this an expensive call and/or not cached? If so, we may want to hoist this up a block and extract all unique `ruleTypes` from the bulk request and only check those (since a user could be importing hundreds of rules at a time). If not, please ignore my premature optimization focus 😅

Reply:
Nah, this should be a simple numeric comparison, ultimately. The (potentially) expensive piece of retrieving the license is already done by the time we're in the route context, here.